pkgs/development/compilers/openjdk/openjfx/11.nix

```diff
@@ -67,47 +67,73 @@ let
-    COMPILE_WEBKIT = true
-  '';
-
-  preBuild = ''
-    swtJar="$(find ${deps} -name org.eclipse.swt\*.jar)"
-    substituteInPlace build.gradle \
-      --replace 'mavenCentral()' 'mavenLocal(); maven { url uri("${deps}") }' \
-      --replace 'name: SWT_FILE_NAME' "files('$swtJar')"
-  '';
-
-  installPhase = ''
-    cp -r build/modular-sdk $out
-  '';
-
-  # glib-2.62 deprecations
-  NIX_CFLAGS_COMPILE = "-DGLIB_DISABLE_DEPRECATION_WARNINGS";
-
-  stripDebugList = [ "." ];
-
-  postFixup = ''
-    # Remove references to bootstrap.
-    find "$out" -name \*.so | while read lib; do
-      new_refs="$(patchelf --print-rpath "$lib" | sed -E 's,:?${openjdk11-bootstrap}[^:]*,,')"
-      patchelf --set-rpath "$new_refs" "$lib"
-    done
-  '';
-
-  disallowedReferences = [ openjdk11-bootstrap ];
-
-  passthru.deps = deps;
-
-  meta = with lib; {
-    homepage = "http://openjdk.java.net/projects/openjfx/";
-    license = licenses.gpl2;
-    description = "The next-generation Java client toolkit";
-    maintainers = with maintainers; [ abbradar ];
-    platforms = [ "i686-linux" "x86_64-linux" ];
+  common = args: makePackage ({
+    patches = [
+      # libglass.so tries to load GTK to see it is available, so we must add GTK
+      # to the RUNPATH of the libglass.so.
+      ./add_gtk_to_libglass_11.patch
+      # libstreamer-lite.so loads FFmpeg at runtime, so we must add FFmpeg to the
+      # RUNPATH of the libstreamer-lite.so.
+      ./add_ffmpeg_to_libgstreamer-lite.patch
+    ];
+
+    gradleProperties = ''
+      COMPILE_MEDIA = true
+      COMPILE_WEBKIT = true
+    '';
+
+    preBuild = ''
+      swtJar="$(find ${deps} -name org.eclipse.swt\*.jar)"
+      substituteInPlace build.gradle \
+        --replace 'mavenCentral()' 'mavenLocal(); maven { url uri("${deps}") }' \
+        --replace 'name: SWT_FILE_NAME' "files('$swtJar')"
+    '';
+
+    stripDebugList = [ "." ];
+
+    postFixup = ''
+      # Remove references to bootstrap.
+      find "$out" -name \*.so | while read lib; do
+        new_refs="$(patchelf --print-rpath "$lib" | sed -E 's,:?${openjdk11-bootstrap}[^:]*,,')"
+        patchelf --set-rpath "$new_refs" "$lib"
+      done
+    '';
+
+    disallowedReferences = [ openjdk11-bootstrap ];
+  } // args);
+
+in {
+  modular-sdk = common {
+    pname = "openjfx-modular-sdk";
+
+    installPhase = ''
+      cp -r build/modular-sdk $out
+    '';
+
+    passthru.deps = deps;
+
+    meta = with lib; {
+      homepage = "http://openjdk.java.net/projects/openjfx/";
+      license = licenses.gpl2;
+      description = "The next-generation Java client toolkit; unpacked SDK";
+      maintainers = with maintainers; [ abbradar ];
+      platforms = [ "i686-linux" "x86_64-linux" ];
+    };
+  };
+
+  sdk = common {
+    pname = "openjfx-sdk";
+
+    installPhase = ''
+      cp -r build/sdk $out
+    '';
+
+    meta = with lib; {
+      homepage = "http://openjdk.java.net/projects/openjfx/";
+      license = licenses.gpl2;
```

Comment:
gpl2Only or gpl2Plus?

Reply:
GPL v2 only, with Classpath exception

https://gluonhq.com/products/javafx/
https://github.com/openjdk/jfx
https://github.com/openjdk/jfx/blob/master/modules/javafx.base/src/main/java/javafx/beans/DefaultProperty.java

Updated the license info.